clippy_lints/src/matches.rs

```diff
@@ -605,12 +598,7 @@ fn check_match_ref_pats(cx: &LateContext<'_, '_>, ex: &Expr, arms: &[Arm], expr:
 }
 
 fn check_match_as_ref(cx: &LateContext<'_, '_>, ex: &Expr, arms: &[Arm], expr: &Expr) {
-    if arms.len() == 2
-        && arms[0].pats.len() == 1
-        && arms[0].guard.is_none()
-        && arms[1].pats.len() == 1
-        && arms[1].guard.is_none()
-    {
+    if arms.len() == 2 && arms[0].guard.is_none() && arms[1].guard.is_none() {
```

Comment:
I feel that I've seen this pattern a few times now btw... maybe refactor in a follow up PR.